lib/plug/crypto/message_verifier.ex

```diff
@@ -127,9 +137,10 @@ defmodule Plug.Crypto.MessageVerifier do
         _ -> String.split(token, "--", parts: 2)
       end
 
-    with  [plain_text, signature] when byte_size(plain_text) > 0 and byte_size(signature) > 0 <- split,
-          {:ok, payload}   <- decode_legacy_base64(plain_text),
-          {:ok, signature} <- decode_legacy_base64(signature) do
+    with [plain_text, signature] when byte_size(plain_text) > 0 and byte_size(signature) > 0 <-
```

Comment:
Not 100% sure about this one, but the `split` in the new line looks a bit weird. Here is a suggestion:

```elixir
with [plain_text, signature] = split,
     true <- byte_size(plain_text) > 0,
     true <- byte_size(signature) > 0,
     {:ok, payload} <- decode_legacy_base64(plain_text),
     {:ok, signature} <- decode_legacy_base64(signature) do
  {"HS1", payload, plain_text, signature}
else
  _ -> :error
end

Reply:
I think we can do `plain_text != ""` :)